src/disc-golf-events/DiscGolfEventsComponent.js

```diff
@@ -29,46 +31,151 @@ const DiscGolfEventsComponent = () => {
     fetchEvents();
   }, [valueToOrderBy, orderDirection]);
 
-  const fetchEvents = () => {
-    let url = `${config.discGolfServiceUrl}/public/events`;
-    if (valueToOrderBy) {
-      url += "?valueToOrderBy=" + valueToOrderBy;
-      if (orderDirection) {
-        url += "&orderDirection=" + orderDirection.toUpperCase();
+  const fetchEvents = async () => {
+    try {
+      let url = `http://localhost:24001/api/disc-golf-service/public/events`;
+      if (valueToOrderBy) {
+        url += `?valueToOrderBy=${valueToOrderBy}`;
+        if (orderDirection) {
+          url += `&orderDirection=${orderDirection.toUpperCase()}`;
+        }
       }
+
+      const response = await fetch(url, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json"
+        },
+        credentials: 'include'
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data = await response.json();
+      setDiscGolfEvents(data);
+      setError(null);
+    } catch (error) {
+      console.error("Error fetching events:", error);
+      setError(error.message);
     }
+  };
 
-    fetch(url)
-      .then((response) => {
-        if (!response.ok) {
-          throw new Error("Received invalid response", response.status);
-        }
-        return response.json();
-      })
-      .then((data) => setDiscGolfEvents(data))
-      .catch((error) => {
-        console.error("Error while fetching events", error);
+  const handleRegister = async (eventId) => {
+    if (!isLoggedIn) {
+      setSnackbar({
+        open: true,
+        message: "Please log in to register for events",
+        severity: "warning"
       });
+      return;
+    }
+
+    try {
+      const response = await fetch(`http://localhost:24001/api/disc-golf-service/events/${eventId}/register`, {
```

Comment:
Let's use config variable instead of localhost, as it won't work on external servers.